File: Set_2/readability.c
#include <stdio.h>
#include <cs50.h>

int length(string s);
void count(string s, int arr[]);

int main(void)
{
    int counts[3];
    string text = get_string("Text: ");
    count(text, counts);
    float L = (float) counts[0] / counts[1] * 100;
    float S = (float) counts[2] / counts[1] * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    //printf("Length: %i\n", length(text));
    //printf("%i letter(s)\n%i word(s)\n%i sentence(s)\n", counts[0], counts[1], counts[2]);
    //printf("L: %f\nS: %f\n", L, S);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", index);
    }
}

int length(string s)
{
    int counter = 0;
    for (int i = 0; s[i] != '\0'; i++)
    {
        counter++;
    }
    return counter;
}

void count(string s, int arr[])
{
    int letters = 0;
    int words = 0;
    int sentences = 0;
    for (int i = 0, n = length(s); i < n; i++)
    {
        if ((s[i] >= 'A' && s[i] <= 'Z') || (s[i] >= 'a' && s[i] <= 'z'))
        {
            letters++;
            if ((s[i + 1] == 0) || (s[i + 1] == 33) || (s[i + 1] == 46) || (s[i + 1] == 63))
            {
                sentences++;
                words++;
            }
            else if ((s[i + 1] == 32) || (s[i + 1] == 34) || (s[i + 1] == 41) || (s[i + 1] == 44) || (s[i + 1] == 58) || (s[i + 1] == 59)
                     || (s[i + 1] == 93) || (s[i + 1] == 125))
            {
                words++;
            }
        }
    }
    arr[0] = letters;
    arr[1] = words;
    arr[2] = sentences;
}

File: Set_6/readability.py
text = input("Text: ")

letters = 0
# Check every object in the user input
# If it is an alphabetical character, add as an increment to the letters var 
for i in text:
    if i.isalpha() == True:
        letters += 1
# print("{} letter(s)".format(letters))

# split the text by space (a default setting of the string.split() method)
# Count the words by the length of the return list
"""Special cases: 
    "sister-in-law" should be considered one word
    "You're" should be considered one word"""
words = len(text.split())
# print("{} word(s)".format(words))

# split the text by '.' or '!' or '?' to count the sentences
"""Special cases: 
    "Mr." should be considered one sentence
    "Mrs." should be considered one sentence"""
sentences = 0
for u in [".", "!", "?"]:
    # the '-1' here at the end is to deduct the last non-used obj in the list
    sentences += len(text.split(u)) - 1
# print(f"{sentences} sentence(s)")

# Coleman-Liau index for calculating levels of grades
index = (0.0588 * letters - 0.296 * sentences) / words * 100 - 15.8

# round the float with the round func by default to the 0 digit after the decimal point.
grade = round(index)

# Some conditions to be eva
if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print("Grade {}".format(grade))